integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java

```diff
@@ -0,0 +1,668 @@
+  @Test
+  @DisplayName("Test domain status condition with serverStartPolicy set to IF_NEEDED")
+  void testMSBootFailureStatus() {
+    try {
+
+      String fmwMiiImage = null;
+
+      String rcuSchemaPrefix = "FMWDOMAINMII";
+      String oracleDbUrlPrefix = "oracledb.";
+      String oracleDbSuffix = null;
+      String rcuSysUserName = "sys";
+      String rcuSysPassword = "Oradoc_db1";
+      String rcuSchemaUserName = "myrcuuser";
+      String rcuSchemaPassword = "Oradoc_db1";
+      String rcuSchemaPasswordNew = "Oradoc_db2";
+      String modelFile = "model-singleclusterdomain-sampleapp-jrf.yaml";
+
+      String rcuaccessSecretName = domainUid + "-rcu-access";
+      String opsswalletpassSecretName = domainUid + "-opss-wallet-password-secret";
+      String opsswalletfileSecretName = domainUid + "opss-wallet-file-secret";
+
+      String dbUrl = oracleDbUrlPrefix + domainNamespace + oracleDbSuffix;
+
+      // create RCU access secret
+      logger.info("Creating RCU access secret: {0}, with prefix: {1}, dbUrl: {2}, schemapassword: {3})",
+          rcuaccessSecretName, rcuSchemaPrefix, rcuSchemaPassword, dbUrl);
+      assertDoesNotThrow(() -> createRcuAccessSecret(rcuaccessSecretName,
+          domainNamespace,
+          rcuSchemaPrefix,
+          rcuSchemaPassword,
+          dbUrl),
+          String.format("createSecret failed for %s", rcuaccessSecretName));
+
+      logger.info("Create OPSS wallet password secret");
+      assertDoesNotThrow(() -> createOpsswalletpasswordSecret(
+          opsswalletpassSecretName,
+          domainNamespace,
+          "welcome1"),
+          String.format("createSecret failed for %s", opsswalletpassSecretName));
+
+      logger.info("Create an image with jrf model file");
+      final List<String> modelList = Collections.singletonList(MODEL_DIR + "/" + modelFile);
+      fmwMiiImage = createMiiImageAndVerify(
+          "jrf-mii-image-status",
+          modelList,
+          Collections.singletonList(MII_BASIC_APP_NAME),
+          FMWINFRA_IMAGE_NAME,
+          FMWINFRA_IMAGE_TAG,
+          "JRF",
+          false);
+
+      // push the image to a registry to make it accessible in multi-node cluster
+      dockerLoginAndPushImageToRegistry(fmwMiiImage);
+
+      // create the domain object
+      Domain domain = FmwUtils.createDomainResource(domainUid,
+          domainNamespace,
+          adminSecretName,
+          OCIR_SECRET_NAME,
+          encryptionSecretName,
+          rcuaccessSecretName,
+          opsswalletpassSecretName,
+          replicaCount,
+          fmwMiiImage);
+
+      createDomainAndVerify(domain, domainNamespace);
+
+      //check the desired completed, available and failed statuses
+      checkStatus("True", "False", "True");
+
+    } finally {
+      deleteDomainResource(domainUid, domainNamespace);
+    }
+  }
+
+  /**
+   * Cleanup the persistent volume and persistent volume claim used by the test.
+   */
+  @AfterAll
+  public static void tearDown() {
```

Comment:
I think the pv and pvc should be cleanup by the framework.

Reply:
removed tearDown